### PR TITLE
Add KACST TrueType Arabic fonts

### PR DESCRIPTION
**Pull request by serge-belov**
###### Description of changes

The package provides KACST Latin-Arabic TrueType fonts (see https://packages.debian.org/bullseye/fonts-kacst)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by serge-belov** [comment]
Closes #166735

File: pkgs/data/fonts/kacst/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchurl, lib }:
+
+stdenv.mkDerivation {
+  name = "kacst-2.01";
```
Comment (wegank) [inline pkgs/data/fonts/kacst/default.nix:4]: Seems better to put 2.01 in the version.

Comment (serge-belov) [reply pkgs/data/fonts/kacst/default.nix:4]: Thanks, I'll fix that

File: pkgs/data/fonts/kacst/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchurl, lib }:
+
+stdenv.mkDerivation {
+  pname = "kacst";
+  version = "2.01";
+
+  src = fetchurl {
+    url = "mirror://debian/pool/main/f/fonts-kacst/fonts-kacst_2.01+mry.orig.tar.bz2";
```
Comment (willjr) [inline pkgs/data/fonts/kacst/default.nix:8]: (suggestion, non-blocking)

A common convention is to replace version tokens with the `${version}` variable.

```suggestion
    url = "mirror://debian/pool/main/f/fonts-kacst/fonts-kacst_${version}+mry.orig.tar.bz2";
```

Comment (serge-belov) [reply pkgs/data/fonts/kacst/default.nix]: I'll investigate the failed check

File: pkgs/data/fonts/kacst/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchurl, lib }:
+
+stdenv.mkDerivation {
```
Comment (willjr) [inline pkgs/data/fonts/kacst/default.nix:3]: This should fix the `ofborg-eval` error:
- https://gist.github.com/d9e9aa333422ecd0d134b69d96859663

```suggestion
stdenv.mkDerivation rec {
```

**Comment by willjr** [comment]
Hi @wegank, just wondering if the fixes we've applied are acceptable?

**Comment by wegank** [comment]
@willjr Should @serge-belov be added as a maintainer?

**Review by wegank — APPROVED** [review]

**Comment by serge-belov** [comment]
> @willjr Should @serge-belov be added as a maintainer?

@wegank I'd love to help the project actually

**Comment by wegank** [comment]
@serge-belov I forgot to mention that the commits should follow the conventions [here](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes). Basically, you need to combine the four commits into one, under the name "kacst: init at 2.01". Then, you can add a second commit adding yourself as a maintainer (see [this](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers)).

**Comment by serge-belov** [comment]
@wegank thanks for your help, will do.

**Comment by wegank** [comment]
@ofborg build kacst

**Comment by willjr** [comment]
> @willjr Should @serge-belov be added as a maintainer?

Sorry for the slow reply.
I think he would be ideal, thanks.

**Review by willjr — APPROVED** [review]

File: pkgs/data/fonts/kacst/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchurl {
+    url = "mirror://debian/pool/main/f/fonts-kacst/fonts-kacst_${version}+mry.orig.tar.bz2";
+    sha256 = "6f2899ce9622314ea426cf8d48849f1cf17302726d73cb683486d5fdf2a23338";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share/fonts/kacst
+    cp kacst/*.ttf $out/share/fonts/kacst
+  '';
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/kacst/default.nix:15]: This could be done in postFetch of fetchurl to remove the extra derivation.

Comment (serge-belov) [reply pkgs/data/fonts/kacst/default.nix:15]: @SuperSandro2000 thanks for your feedback, I'll change it.

**Pull request merged**